### PR TITLE
Aws cloud formation first integration

### PR DESCRIPTION
**Pull request by lrevnic**

### CODE REVIEW

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/CloudFormationClientFactory.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*
```
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/CloudFormationClientFactory.java]: IMHO this should stay in factory (separation of concerns) ... probably CloudFormationClientBuilder named ... and it should follow the builder pattern (private constructor and public static inner class as builder) ... but I can live with it...

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/CloudFormationClientFactory.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            String secretAccessKey,
+            String proxyHost,
+            String proxyPort,
+            String proxyUsername,
+            String proxyPassword,
+            String connectTimeoutMs,
+            String executionTimeoutMs,
+            String region) {
+
+        ClientConfiguration clientConf = new ClientConfiguration()
```
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/CloudFormationClientFactory.java]: It's hard to follow in this way...

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/CreateStackAction.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+        try {
+            AmazonCloudFormation stackBuilder = CloudFormationClientFactory.getCloudFormationClient(identity, credential, proxyHost, proxyPort, proxyUsername, proxyPassword, connectTimeoutMs, execTimeoutMs, region);
+
+            // Create a stack
+            CreateStackRequest createRequest = new CreateStackRequest();
+            createRequest.setStackName(stackName);
+            createRequest.setTemplateBody(templateBody);
+
+            CreateStackResult result = stackBuilder.createStack(createRequest);
```
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/CreateStackAction.java]: It shouldn't be here... maybe in some service class... WDYT?

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/ListStacksAction.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            @Param(value = EXECUTION_TIMEOUT)            String execTimeoutMs) {
+
+        proxyPort = defaultIfEmpty(proxyPort, DefaultValues.PROXY_PORT);
+        connectTimeoutMs = defaultIfEmpty(connectTimeoutMs, DefaultValues.CONNECT_TIMEOUT);
+        execTimeoutMs = defaultIfBlank(execTimeoutMs, DefaultValues.EXEC_TIMEOUT);
+
+        try {
+            AmazonCloudFormation stackBuilder = CloudFormationClientFactory.getCloudFormationClient(identity, credential, proxyHost, proxyPort, proxyUsername, proxyPassword, connectTimeoutMs, execTimeoutMs, region);
+
+            StringBuilder listOfStacksResult = new StringBuilder(EMPTY);
```
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/ListStacksAction.java]: All the following logic shouldn't be here... A specialized service with some kind of wrapper it would be better, right?

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/CreateStackAction.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+import com.hp.oo.sdk.content.plugin.ActionMetadata.ResponseType;
+import io.cloudslang.content.amazon.entities.constants.Outputs;
+import io.cloudslang.content.amazon.utils.DefaultValues;
+import io.cloudslang.content.utils.OutputUtilities;
+
+import java.util.Map;
+
+import static io.cloudslang.content.amazon.entities.constants.Inputs.CloudFormationInputs.STACK_NAME;
+import static io.cloudslang.content.amazon.entities.constants.Inputs.CloudFormationInputs.TEMPLATE_BODY;
+import static io.cloudslang.content.amazon.entities.constants.Inputs.CommonInputs.*;
```
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/CreateStackAction.java]: I would try to avoid (as much as possible) wildcarded imports... Pick just what is needed...

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/ListStacksAction.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+import com.hp.oo.sdk.content.plugin.ActionMetadata.MatchType;
+import com.hp.oo.sdk.content.plugin.ActionMetadata.ResponseType;
+import io.cloudslang.content.amazon.entities.constants.Outputs;
+import io.cloudslang.content.amazon.utils.DefaultValues;
+import io.cloudslang.content.utils.OutputUtilities;
+
+import java.util.Map;
+
+import static io.cloudslang.content.amazon.entities.constants.Inputs.CommonInputs.*;
+import static org.apache.commons.lang3.StringUtils.*;
```
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/ListStacksAction.java]: Same comment here... especially when third party library is involved. It doesn't make sense to import it all... just method / methods needed. Minimal I mean... :smile:

**Comment by mihaitusa** [comment]
Great job @lrevnic ... A new fresh Amazon approach... maybe will get together some time... I'll want to contribute here to...

**Comment by mihaitusa** [comment]
Can cs-actions made to work with Java 8? It's a unrelated with this PR question but... a valid one...

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/CreateStackAction.java (hunk 5 above)
Comment (lrevnic) [reply cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/CreateStackAction.java]: Done!

**Comment by lrevnic** [comment]
@mihaitusa thanks for your comments! 
We are assessing to migrate to java8 but not yet officially supported. In few months from now it will be.
